src/System.Runtime.Analyzers/Core/SpecifyIFormatProvider.cs

```diff
@@ -27,42 +34,203 @@ public abstract class SpecifyIFormatProviderAnalyzer : DiagnosticAnalyzer
+                var stringFormatMembers = stringType?.GetMembers("Format").OfType<IMethodSymbol>();
+                var stringFormatMemberWithStringAndObjectParameter = GetSingleOrDefaultMemberWithName(stringFormatMembers, "string.Format(string, object)");
+                var stringFormatMemberWithStringObjectAndObjectParameter = GetSingleOrDefaultMemberWithName(stringFormatMembers, "string.Format(string, object, object)");
+                var stringFormatMemberWithStringObjectObjectAndObjectParameter = GetSingleOrDefaultMemberWithName(stringFormatMembers, "string.Format(string, object, object, object)");
+
+                var paramsObjectArraySymbolDisplay = csaContext.Compilation.Language == LanguageNames.CSharp ?
+                                                     "params object[]" :
+                                                     "ParamArray Object()";
+                var stringFormatMemberWithStringAndParamsObjectParameter = GetSingleOrDefaultMemberWithName(stringFormatMembers, $"string.Format(string, {paramsObjectArraySymbolDisplay})");
+                var stringFormatMemberWithIFormatProviderStringAndParamsObjectParameter = GetSingleOrDefaultMemberWithName(stringFormatMembers, $"string.Format(System.IFormatProvider, string, {paramsObjectArraySymbolDisplay})");
+
+                var IFormatProviderType = csaContext.Compilation.GetTypeByMetadataName("System.IFormatProvider");
+
+                var CultureInfoType = csaContext.Compilation.GetTypeByMetadataName("System.Globalization.CultureInfo");
+                var currentCultureProperty = CultureInfoType?.GetMembers("CurrentCulture").OfType<IPropertySymbol>().SingleOrDefault();
+                var invariantCultureProperty = CultureInfoType?.GetMembers("InvariantCulture").OfType<IPropertySymbol>().SingleOrDefault();
+                var currentUICultureProperty = CultureInfoType?.GetMembers("CurrentUICulture").OfType<IPropertySymbol>().SingleOrDefault();
+                var installedUICultureProperty = CultureInfoType?.GetMembers("InstalledUICulture").OfType<IPropertySymbol>().SingleOrDefault();
+
+                var threadType = csaContext.Compilation.GetTypeByMetadataName("System.Threading.Thread");
+                var currentThreadCurrentUICultureProperty = threadType?.GetMembers("CurrentUICulture").OfType<IPropertySymbol>().SingleOrDefault();
+
+                var activatorType = csaContext.Compilation.GetTypeByMetadataName("System.Activator");
+                var resourceManagerType = csaContext.Compilation.GetTypeByMetadataName("System.Resources.ResourceManager");
+
+                var computerInfoType = csaContext.Compilation.GetTypeByMetadataName("Microsoft.VisualBasic.Devices.ComputerInfo");
+                var installedUICulturePropertyOfComputerInfoType = computerInfoType?.GetMembers("InstalledUICulture").OfType<IPropertySymbol>().SingleOrDefault();
+                #endregion
+
```

Comment:
Is it possible to skip doing any analysis if any of these types are not present? Or are most of the types only needed for corner cases?